linmod/tests/test_eval.py

```diff
@@ -0,0 +1,345 @@
+        phi=pl.Series(
+            [
+                0.2,
+                0.8,
+                0.3,
+                0.7,
+                0.1,
+                0.9,
+                0.4,
+                0.6,
+                0.5,
+                0.5,
+                0.1,
+                0.9,
+                0.2,
+                0.8,
+                0.3,
+                0.7,
+                0.1,
+                0.9,
+                0.4,
+                0.6,
+                0.5,
+                0.5,
+                0.1,
+                0.9,
+            ]
+        ),
+    )
+
+    correct_output = pl.DataFrame(
+        {
+            "fd_offset": [0, 0, 1, 1],
+            "division": [0, 1, 0, 1],
+            "energy_score": [
+                np.abs(np.array([0.2, 0.3, 0.1]) - 1 / (1 + 2)).mean()
+                + np.abs(np.array([0.8, 0.7, 0.9]) - 2 / (1 + 2)).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.2, 0.3, 0.1]) - np.array([0.1, 0.2, 0.3])
+                ).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.8, 0.7, 0.9]) - np.array([0.9, 0.8, 0.7])
+                ).mean(),
+                np.abs(np.array([0.4, 0.5, 0.1]) - 3 / (3 + 4)).mean()
+                + np.abs(np.array([0.6, 0.5, 0.9]) - 4 / (3 + 4)).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.4, 0.5, 0.1]) - np.array([0.1, 0.4, 0.5])
+                ).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.6, 0.5, 0.9]) - np.array([0.9, 0.6, 0.5])
+                ).mean(),
+                np.abs(np.array([0.2, 0.3, 0.1]) - 5 / (5 + 6)).mean()
+                + np.abs(np.array([0.8, 0.7, 0.9]) - 6 / (5 + 6)).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.2, 0.3, 0.1]) - np.array([0.1, 0.2, 0.3])
+                ).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.8, 0.7, 0.9]) - np.array([0.9, 0.8, 0.7])
+                ).mean(),
+                np.abs(np.array([0.4, 0.5, 0.1]) - 7 / (7 + 8)).mean()
+                + np.abs(np.array([0.6, 0.5, 0.9]) - 8 / (7 + 8)).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.4, 0.5, 0.1]) - np.array([0.1, 0.4, 0.5])
+                ).mean()
+                - 0.5
+                * np.abs(
+                    np.array([0.6, 0.5, 0.9]) - np.array([0.9, 0.6, 0.5])
+                ).mean(),
+            ],
+        }
+    )
+
+    result = eval.proportions_energy_score_per_division_day(
+        samples, data, p=1
+    ).collect()
+
+    assert_frame_equal(
+        result,
+        correct_output,
+        check_row_order=False,
+        check_column_order=True,
+    )
+
+
+test_proportions_mean_L1_norm()
+test_proportions_L1_energy_score()
+test_proportions_mean_L1_norm2()
+test_proportions_L1_energy_score2()
```

Comment:
I'm like 95% sure we don't need these and that `pytest` will run all defined `test_[blah]` functions automatically